lib/streamlit/elements/time_widgets.py

```diff
@@ -102,9 +104,11 @@ def date_input(
             0 to 2 date/datetime values is provided, the datepicker will allow
             users to provide a range. Defaults to today as a single-date picker.
         min_value : datetime.date or datetime.datetime
-            The minimum selectable date. Defaults to today-10y.
+            The minimum selectable date. Defaults to value - 10 years or left
+            boundary of interval - 10 years, if value is а range.
```

Comment:
The documentation should be updated automatically (as long as this docstring is already in the existing docs, [which it is](https://docs.streamlit.io/en/stable/api.html#streamlit.date_input)) since docs in our API reference are rendered from the docstrings themselves.

Reply:
As for the documentation wording, I think this is fine, but it may be slightly clearer to phrase it as something like

"The minimum selectable date. If value is a date, defaults to value - 10 years. If value is the interval [start, end], defaults to start - 10 years."